File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/data_structures/test_vector.java
// test_vector.java

public class test_vector
{
	public static <T extends Object> void display_vector(dp_vector<T> a_vector)
	{
		System.out.print("display_vector(a_vector): <front> ");
		dp_iterator<T> a_iterator = a_vector.make_iterator();
		while (!a_iterator.is_done())
		{
			T a_value = a_iterator.get_current();
			System.out.print(a_value + " ");
			a_iterator.advance();
		}
		System.out.println("<back> (size = " + a_vector.get_size() + ")");
	}

	public static <T extends Object> void display_reverse_vector(dp_vector<T> a_vector)
	{
		System.out.print("display_vector(a_vector): <front> ");
		dp_iterator<T> a_iterator = a_vector.make_reverse_iterator();
		while (!a_iterator.is_done())
		{
			T a_value = a_iterator.get_current();
			System.out.print(a_value + " ");
			a_iterator.advance();
		}
		System.out.println("<back> (size = " + a_vector.get_size() + ")");
	}

	public static void main(String[] args)
	{
		dp_vector<Integer> a_vector = new dp_vector<Integer>();
		display_vector(a_vector);
		display_reverse_vector(a_vector);

		a_vector.add(1);
		display_vector(a_vector);
		display_reverse_vector(a_vector);

		a_vector.add(2);
		display_vector(a_vector);
		display_reverse_vector(a_vector);
	}
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_matlab_labs_and_solutions/data_structures/testVector.m
% testVector.m

function testVector
	aVector = DpVector;
	displayVector(aVector);
	displayReverseVector(aVector);

	aVector.add(1);
	displayVector(aVector);
	displayReverseVector(aVector);

	aVector.add(2);
	displayVector(aVector);
	displayReverseVector(aVector);
end

function displayVector(aVector)
	fprintf('displayVector(aVector): <front> ');
	aIterator = aVector.makeIterator;
	while ~aIterator.isDone
		aValue = aIterator.getCurrent;
		fprintf('%d ', aValue);
		aIterator.advance;
	end
	fprintf('<back> (size = %d)\n', aVector.getSize);
end

function displayReverseVector(aVector)
	fprintf('displayReverseVector(aVector): <back> ');
	aIterator = aVector.makeReverseIterator;
	while ~aIterator.isDone
		aValue = aIterator.getCurrent;
		fprintf('%d ', aValue);
		aIterator.advance;
	end
	fprintf('<front> (size = %d)\n', aVector.getSize);
end
